File: initialstate.f
      PROGRAM ISTATE
      IMPLICIT NONE
      INTEGER STRLEN
      PARAMETER (STRLEN=32)
      CHARACTER*(STRLEN) UTC0
      CHARACTER*(STRLEN) ABCORR
      CHARACTER*(STRLEN) FRAME
      CHARACTER*7 BODIES(0:4)
      DOUBLE PRECISION GM,STATE(6),ET0,JD0,LT
      INTEGER I,N
      DATA BODIES/'SUN','VENUS','EARTH','JUPITER','CASSINI'/
C
      CALL FURNSH('cassini.meta')
C
      CALL PROMPT('Enter the beginning UTC time: ', UTC0)
      CALL STR2ET(UTC0, ET0)
C
      OPEN(UNIT=8,FILE='initialstate.out',STATUS='UNKNOWN')
C
      JD0=2451545.0D0+ET0/86400.0D0
      WRITE(8,1000) JD0
 1000 FORMAT(F15.6)
C
      DO I=0,3
        CALL BODVRD(BODIES(I) , 'GM', 1, N, GM )
        WRITE(8,1001) GM
      END DO
 1001 FORMAT(D20.12)
C
      ABCORR='NONE'
      FRAME='J2000'
C
      DO I=1,4
        CALL SPKEZR(BODIES(I), ET0, FRAME, ABCORR, 'SUN', STATE, LT)
        WRITE(8,1002) STATE
      END DO
 1002 FORMAT(6(1X,D20.12))
C
      CLOSE(UNIT=8,STATUS='KEEP')
C
      CALL BYEBYE( 'SUCCESS' )
      END

File: initialstate.py
#!/usr/bin/env python3

import spiceypy as spice
import sys
from pathlib import Path
import glob

if len(sys.argv) < 2:
    print("Usage: ", sys.argv[0], " start-date")
    quit()

kernels=str(Path.home()) + '/kernels/*.*'

for kernel in glob.glob(kernels):
    spice.furnsh(kernel)

t=spice.str2et(sys.argv[1])

for target in ['SUN','VENUS','EARTH','JUPITER','SATURN']:
    p=spice.bodvrd(target, 'GM', 1)
    gm=p[1][0]
    print(target,' ',gm)

for target in ['VENUS BARYCENTER','EARTH BARYCENTER','JUPITER BARYCENTER','SATURN BARYCENTER','CASSINI']:
    state=spice.spkezr(target,t,'J2000','NONE','SUN')
    print(target,' ',state[0])
